Extensions/Security/Basic.md

```diff
@@ -0,0 +1,29 @@
+# Basic Authentication Extension
+
+_This extension specification is currently incubating.  While incubating the version is 0._
+
+## Introduction
+Authentication is a necessary component to any real world application. The most "basic" mechanism for authenticating is a leveraging a username and password for authentication. This extension specification provides a standardized mechanism for including a username and password in metadata payloads.
+
+## Metadata Payload
+This metadata type can be used in a per connection or per stream, and not individual payloads and as such it **MUST** only be used in frame types used to initiate interactions and payloads.  This includes [`SETUP`][s], [`REQUEST_FNF`][rf], [`REQUEST_RESPONSE`][rr], [`REQUEST_STREAM`][rs], [`REQUEST_CHANNEL`][rc], and [`PAYLOAD`][p].  The Metadata MIME Type is `message/x.rsocket.authentication.basic.v0`.
+
+[s]:  ../Protocol.md#frame-setup
+[p]:  ../Protocol.md#frame-payload
+[rc]: ../Protocol.md#frame-request-channel
+[rf]: ../Protocol.md#frame-fnf
+[rr]: ../Protocol.md#frame-request-response
+[rs]: ../Protocol.md#frame-request-stream
+
+### Metadata Contents
+```
+     0                   1                   2                   3
+     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |Username Length|     Username  |    Password              ...
+    +---------------+-----------------------------------------------+
+```
+
+* **Username Length**: Username Length in bytes.
+* **Username**:  The UTF-8 encoded username.  The string MUST NOT be null terminated.
+* **Password**:  The UTF-8 encoded password.  The string MUST NOT be null terminated.
```

Comment:
Password Length?

Reply:
How about tagging for username and password?  https://github.com/rsocket/rsocket/blob/master/Extensions/Routing.md .  The first tag is user name, and the second is password?

Reply:
> Password Length?

Isn't that just the remaining bytes?